Content.Server/Mech/Equipment/EntitySystems/MechGrabberSystem.cs

```diff
@@ -0,0 +1,173 @@
+            return;
+        var mech = equipmentComponent.EquipmentOwner.Value;
+
+        var targetCoords = new EntityCoordinates(mech, component.DepositOffset);
+        if (!_interaction.InRangeUnobstructed(mech, targetCoords))
+            return;
+
+        if (!component.ItemContainer.Contains(msg.Item))
+            return;
+
+        Logger.Debug("ok now remove it");
+        RemoveItem(uid, mech, msg.Item, component);
+    }
+
+    public void RemoveItem(EntityUid uid, EntityUid mech, EntityUid toRemove, MechGrabberComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        component.ItemContainer.Remove(toRemove);
+        var mechxform = Transform(mech);
+        var xform = Transform(toRemove);
+        xform.AttachToGridOrMap();
+        xform.WorldPosition = mechxform.WorldPosition + mechxform.WorldRotation.RotateVec(component.DepositOffset);
+        xform.WorldRotation = Angle.Zero;
+        _mech.UpdateUserInterface(mech);
+    }
+
+    private void OnEquipmentRemoved(EntityUid uid, MechGrabberComponent component, ref MechEquipmentRemovedEvent args)
+    {
+        if (!TryComp<MechEquipmentComponent>(uid, out var equipmentComponent) ||
+            equipmentComponent.EquipmentOwner == null)
+            return;
+        var mech = equipmentComponent.EquipmentOwner.Value;
+
+        var allItems = new List<EntityUid>(component.ItemContainer.ContainedEntities);
+        foreach (var item in allItems)
+        {
+            RemoveItem(uid, mech, item, component);
+        }
+    }
+
+    private void OnAttemptRemove(EntityUid uid, MechGrabberComponent component, ref AttemptRemoveMechEquipmentEvent args)
+    {
+        args.Cancelled = component.ItemContainer.ContainedEntities.Any();
+    }
+
+    private void OnStartup(EntityUid uid, MechGrabberComponent component, ComponentStartup args)
+    {
+        component.ItemContainer = _container.EnsureContainer<Container>(uid, "item-container");
+    }
+
+    private void OnUiStateReady(EntityUid uid, MechGrabberComponent component, MechEquipmentUiStateReadyEvent args)
+    {
+        var state = new MechGrabberUiState
+        {
+            Contents = component.ItemContainer.ContainedEntities.ToList(),
+            MaxContents = component.MaxContents
+        };
+        args.States.Add(uid, state);
+    }
+
+    private void OnInteract(EntityUid uid, MechGrabberComponent component, InteractNoHandEvent args)
+    {
+        if (args.Handled || args.Target == null)
+            return;
+
+        var xform = Transform(args.Target.Value);
+        if (xform.Anchored || HasComp<WallMountComponent>(args.Target.Value))
+            return;
+
+        if (component.ItemContainer.ContainedEntities.Count >= component.MaxContents)
+            return;
+
+        if (!TryComp<MechComponent>(args.User, out var mech))
+            return;
+
+        if (mech.Energy + component.EnergyPerGrab < 0)
```

Comment:
Is this meant to be a subtraction?

Reply:
no. The energy per grab is a negative number. I'll rename it to be a bit clearer